pydash/pydash_app/dashboard/aggregator/statistics.py

```diff
@@ -136,3 +134,104 @@ def append(self, endpoint_call, dependencies):
 
     def rendered_value(self):
         return date_dict({k: len(v) for k, v in self.value.items()})
+
+
+class FastestExecutionTime(Statistic):
+    dependencies = []
+
+    def empty(self):
+        return -1
+
+    def field_name(self):
+        return 'fastest_measured_execution_time'
+
+    def append(self, endpoint_call, dependencies):
+        if self.value < 0 or self.value > endpoint_call.execution_time:
+            self.value = endpoint_call.execution_time
+
+    def rendered_value(self):
+        return self.value
+
+
+class SlowestExecutionTime(Statistic):
+    dependencies = []
+
+    def empty(self):
+        return -1
+
+    def field_name(self):
+        return 'slowest_measured_execution_time'
+
+    def append(self, endpoint_call, dependencies):
+        if self.value < endpoint_call.execution_time:
+            self.value = endpoint_call.execution_time
+
+    def rendered_value(self):
+        return self.value
+
+
+class TDigestStatistic(Statistic):
+    dependencies = []
+
+    def __init__(self):
+        super().__init__()
+        self.value = self.empty()
+        self._digest = tdigest.TDigest()
+
+    def empty(self):
+        return -1
+
+    @abc.abstractmethod
+    def field_name(self):
+        pass
+
+    @abc.abstractmethod
+    def append(self, endpoint_call, dependencies):
+        self._digest.update(endpoint_call.execution_time)
+
+    def rendered_value(self):
+        return self.value
+
+
+class FastestQuartileExecutionTime(TDigestStatistic):
+    dependencies = []
+
+    def field_name(self):
+        return 'fastest_quartile_execution_time'
+
+    def append(self, endpoint_call, dependencies):
+        super().append(endpoint_call, dependencies)
```

Comment:
Rather than each of these statistics having their own TDigest data type, it would be a lot nicer if you'd have a single TDigest statistic, with these others returning their value from reading that one (using it as a dependency).

(you perform a lot of calculations multiple times now)

Reply:
Indeed, I had not implemented it that way because I thought that such dependencies would also be added to the `as_dict` return value. Good point though, I will implement that after the meeting.